File: Models/RoleplayResult.cs
namespace JCAP.Models
{
    public class RoleplayResult
    {
        public int Id { get; set; }
        public int RoleplaySessionId { get; set; }
        public string UserId { get; set; } = string.Empty;
        public string ScenarioTitle { get; set; } = string.Empty;
        public string JLPTLevel { get; set; } = string.Empty;
        public int OverallScore { get; set; }
        public int GrammarScore { get; set; }
        public int VocabularyScore { get; set; }
        public int ImpressionScore { get; set; }
        public bool PassStatus { get; set; }
        public string GeneralFeedbackText { get; set; } = string.Empty;
        public string CompletedMissionsSummaryJson { get; set; } = "[]";
        public DateTime CompletedAt { get; set; } = DateTime.UtcNow;
        public ApplicationUser? User { get; set; }
    }
}

File: jcap-web/src/types/roleplayResult.ts
export interface CompleteRoleplaySessionResponse {
  resultId: number;
  isExistingResult: boolean;
  isMockEvaluation: boolean;
}

export interface CompletedMissionResult {
  missionId: number;
  title: string;
}

export interface RoleplayResultSummary {
  id: number;
  roleplaySessionId: number;
  scenarioTitle: string;
  jlptLevel: string;
  overallScore: number;
  passStatus: boolean;
  completedAt: string;
}

export interface RoleplayResultDetail extends RoleplayResultSummary {
  grammarScore: number;
  vocabularyScore: number;
  impressionScore: number;
  generalFeedbackText: string;
  completedMissions: CompletedMissionResult[];
}

export interface RoleplayResultHistoryResponse {
  items: RoleplayResultSummary[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}
